unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAnimationEventHandler.cs

```diff
@@ -0,0 +1,158 @@
+        footstepSlide = audioContainer.GetEvent("FootstepSlide");
+        footstepWalk = audioContainer.GetEvent("FootstepWalk");
+        footstepRun = audioContainer.GetEvent("FootstepRun");
+        clothesRustleShort = audioContainer.GetEvent("ClothesRustleShort");
+        clap = audioContainer.GetEvent("ExpressionClap");
+        throwMoney = audioContainer.GetEvent("ExpressionThrowMoney");
+        blowKiss = audioContainer.GetEvent("ExpressionBlowKiss");
+
+        bodyPartReferenceHandler = GetComponent<AvatarBodyPartReferenceHandler>();
+        anim = GetComponent<Animation>();
+        stickersController = GetComponentInParent<StickersController>();
+    }
+
+    public void AnimEvent_FootstepLight() { PlayAudioEvent(footstepLight); }
+
+    public void AnimEvent_FootstepSlide() { PlayAudioEvent(footstepSlide); }
+
+    public void AnimEvent_FootstepWalkLeft() { PlayAudioEvent(footstepWalk); }
+
+    public void AnimEvent_FootstepWalkRight() { PlayAudioEvent(footstepWalk); }
+
+    public void AnimEvent_FootstepRunLeft()
+    {
+        PlayAudioEvent(footstepRun);
+        PlaySticker("footstep", bodyPartReferenceHandler.footL.position, Vector3.up);
+    }
+
+    public void AnimEvent_FootstepRunRight()
+    {
+        PlayAudioEvent(footstepRun);
+        PlaySticker("footstep", bodyPartReferenceHandler.footR.position, Vector3.up);
+    }
+
+    public void AnimEvent_ClothesRustleShort() { PlayAudioEvent(clothesRustleShort); }
+
+    public void AnimEvent_Clap()
+    {
+        if (LastEventWasTooRecent())
+            return;
+
+        if (!AnimationWeightIsOverThreshold(0.2f, ANIM_NAME_CLAP))
+            return;
+
+        PlayAudioEvent(clap);
+        PlaySticker("clap", bodyPartReferenceHandler.handR.position, Vector3.up);
+        UpdateEventTime();
+    }
+
+    public void AnimEvent_ThrowMoney()
+    {
+        if (LastEventWasTooRecent())
+            return;
+
+        if (!AnimationWeightIsOverThreshold(0.2f, ANIM_NAME_MONEY))
+            return;
+
+        PlayAudioEvent(throwMoney);
+        PlaySticker("money", bodyPartReferenceHandler.handL.position, bodyPartReferenceHandler.handL.rotation.eulerAngles);
+        UpdateEventTime();
+    }
+
+    public void AnimEvent_BlowKiss()
+    {
+        if (LastEventWasTooRecent())
+            return;
+
+        if (!AnimationWeightIsOverThreshold(0.2f, ANIM_NAME_KISS))
+            return;
+
+        PlayAudioEvent(blowKiss);
+        StartCoroutine(EmitHeartParticle());
+        UpdateEventTime();
+    }
+
+    IEnumerator EmitHeartParticle()
+    {
+        yield return new WaitForSeconds(0.8f);
+        PlaySticker("heart", bodyPartReferenceHandler.handR.position, transform.rotation.eulerAngles);
+    }
+
+    void PlayAudioEvent(AudioEvent audioEvent)
+    {
+        if (audioEvent != null)
+            audioEvent.Play(true);
+    }
+
+    bool AnimationWeightIsOverThreshold(float threshold, string animationName)
+    {
+        if (anim != null) {
```

Comment:
```suggestion
        if (anim != null) 
        {
```
New line before brackets